nvflare/app_common/ccwf/ccwf_job.py

```diff
@@ -0,0 +1,308 @@
+# limitations under the License.
+from typing import List, Optional
+
+from nvflare.apis.executor import Executor
+from nvflare.app_common.abstract.aggregator import Aggregator
+from nvflare.app_common.abstract.metric_comparator import MetricComparator
+from nvflare.app_common.abstract.model_persistor import ModelPersistor
+from nvflare.app_common.abstract.shareable_generator import ShareableGenerator
+from nvflare.app_common.app_constant import AppConstants
+from nvflare.app_common.ccwf.common import Constant, CyclicOrder
+from nvflare.fuel.utils.validation_utils import check_object_type
+from nvflare.job_config.api import ControllerApp, ExecutorApp, FedJob
+from nvflare.widgets.widget import Widget
+
+from .cse_client_ctl import CrossSiteEvalClientController
+from .cse_server_ctl import CrossSiteEvalServerController
+from .cyclic_client_ctl import CyclicClientController
+from .cyclic_server_ctl import CyclicServerController
+from .swarm_client_ctl import SwarmClientController
+from .swarm_server_ctl import SwarmServerController
+
+
+class FedAvgServerConfig:
+    def __init__(
+        self,
+        num_rounds: int,
+        start_round: int = 0,
+        start_task_timeout=Constant.START_TASK_TIMEOUT,
+        configure_task_timeout=Constant.CONFIG_TASK_TIMEOUT,
+        participating_clients=None,
+        result_clients=None,
+        starting_client: str = "",
+        max_status_report_interval: float = Constant.PER_CLIENT_STATUS_REPORT_TIMEOUT,
+        progress_timeout: float = Constant.WORKFLOW_PROGRESS_TIMEOUT,
+        private_p2p: bool = True,
+        aggr_clients=None,
+        train_clients=None,
+    ):
+        self.num_rounds = num_rounds
+        self.start_round = start_round
+        self.start_task_timeout = start_task_timeout
+        self.configure_task_timeout = configure_task_timeout
+        self.participating_clients = participating_clients
+        self.result_clients = result_clients
+        self.starting_client = starting_client
+        self.max_status_report_interval = max_status_report_interval
+        self.progress_timeout = progress_timeout
+        self.private_p2p = private_p2p
+        self.aggr_clients = aggr_clients
+        self.train_clients = train_clients
+
+
+class FedAvgClientConfig:
+    def __init__(
+        self,
+        executor: Executor,
+        persistor: ModelPersistor,
+        shareable_generator: ShareableGenerator,
+        aggregator: Aggregator,
+        metric_comparator: MetricComparator = None,
+        model_selector: Widget = None,
+        learn_task_check_interval=Constant.LEARN_TASK_CHECK_INTERVAL,
+        learn_task_abort_timeout=Constant.LEARN_TASK_ABORT_TIMEOUT,
+        learn_task_ack_timeout=Constant.LEARN_TASK_ACK_TIMEOUT,
+        learn_task_timeout=None,
+        final_result_ack_timeout=Constant.FINAL_RESULT_ACK_TIMEOUT,
+        min_responses_required: int = 1,
+        wait_time_after_min_resps_received: float = 10.0,
+    ):
```

Comment:
since this object is user facing, could we add docstring to it and explain the arguments?